src/octopeer-github/content/ContentController.ts

```diff
@@ -88,22 +87,54 @@ class Controller {
         FilesChangedTabHeaderElementSelectionBehaviour,
     ];
 
-    /** Starts the Controller. After calling this, all event handlers are hooked to the DOM-tree. */
+    /**
+     * Starts the ContentController. After calling this, all event handlers are hooked to the DOM-tree.
+     * @return this
+     */
     public start() {
-        this.database = new DatabaseConsoleLogOnly(); // ("https://localhost:8000", 1, 1);
-        this.hookToDOM();
+        this.connectToBackgroundPage();
         return this;
     }
 
-    /** Hook the product of ElementBindings and ElementSelectors to the DOM-tree. */
-    private hookToDOM() {
+    /**
+     * Set up all event handlers in the Chrome API.
+     */
+    private connectToBackgroundPage() {
+        const self = this;
+        if (chrome.runtime.onMessage.hasListeners()) {
+            return;
+        }
+
+        chrome.runtime.onMessage.addListener(function (request, sender, sendResponse) {
+            if (request.hookToDom) {
+                self.hookToDOM({
+                    post: function (data: EventObject, success: Callback, failure: Callback) {
```

Comment:
`failure` is part of the Interface of `DatabaseAdaptable` and should be called whenever the post to the database failed :) I'd better document that :P